rke/templates/nginx-ingress.go

```diff
@@ -612,9 +616,15 @@ spec:
 {{end}}
           ports:
           - name: http
-            containerPort: 80
+            containerPort: {{with (index .ExtraArgs "http-port")}}{{.}}{{else}}80{{end}}
+            {{- if eq .HostNetwork "enabled" }}
+            hostPort: {{if .HttpPort}}{{.HttpPort}}{{else}}80{{end}}
```

Comment:
When using `hostNetwork: true` the process binds directly on the host, but using host ports, we would have iptables doing the forwarding. We might want to keep the existing behavior @ibrokethecloud?

Reply:
The reason for hostPort was that we wanted the user to know exactly what they were doing.

For example, the user can specify a different port for the container to listen on and bind on a host to a different one.

There were quite a few combinations, and I had initially done the `hostNetwork: true` but based on feedback I had tweaked it to `hostPort` to allow users more flexbility.